crates/rust-analyzer/src/config.rs

```diff
@@ -1219,30 +1311,41 @@ enum ManifestOrProjectJson {
 #[derive(Deserialize, Debug, Clone)]
 #[serde(rename_all = "snake_case")]
 pub enum ExprFillDefaultDef {
-    #[serde(alias = "todo")]
     Todo,
-    #[serde(alias = "default")]
     Default,
 }
 
 #[derive(Deserialize, Debug, Clone)]
 #[serde(rename_all = "snake_case")]
 enum ImportGranularityDef {
     Preserve,
-    #[serde(alias = "none")]
     Item,
-    #[serde(alias = "full")]
     Crate,
-    #[serde(alias = "last")]
     Module,
 }
 
 #[derive(Deserialize, Debug, Clone)]
 #[serde(rename_all = "snake_case")]
+enum CallableCompletionDef {
+    FillArguments,
+    AddParentheses,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+#[serde(untagged)]
+enum CargoFeatures {
+    All,
+    Listed(Vec<String>),
+}
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+#[serde(untagged)]
 enum LifetimeElisionDef {
-    #[serde(alias = "true")]
+    #[serde(deserialize_with = "true_or_always")]
     Always,
-    #[serde(alias = "false")]
+    #[serde(deserialize_with = "false_or_never")]
     Never,
     SkipTrivial,
 }
```

Comment:
These don't seem to work as I expected them to regarding serde, will have to look into that (untagged enums + unit variants that is)

Reply:
ah well that explains it https://github.com/serde-rs/serde/issues/1560